test/functional/feature_startupnotify.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test -startupnotify."""
+
+import platform
+import os
+
+from test_framework.test_framework import BitcoinTestFramework
+
+INVALID_COMMAND = "abc"
+
+
+class StartupNotifyTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.disable_syscall_sandbox = True
+        if platform.system() == "Windows":
+            self.extra_args = [["-startupnotify=dir"]]
+        else:
+            self.extra_args = [["-startupnotify=ls"]]
+
+    def run_test(self):
+        self.log.info("Test the node starts with a valid command on -startupnotify")
+        self.stop_node(0)
+        self.log.info("Test the node starts with an invalid command on -startupnotify")
+        self.start_node(0, extra_args=[f"-startupnotify={INVALID_COMMAND}"])
+
+        stderrs = [f"sh: {INVALID_COMMAND}: command not found", f"sh: 1: {INVALID_COMMAND}: not found",
+        f"AssertionError: Unexpected stderr '{INVALID_COMMAND}' is not recognized as an internal or external command,{os.linesep}operable program or batch file."]
```

Comment:
Do we need tests for to check the operating systems exact response to an invalid command? This doesn't increase the test coverage of Bitcoin Core itself and it seems a bit painful to maintain.

Reply:
I agree it seems a bit painful to maintain, but if we don't pass an expected error message the test will fail. I think the stop_node could have a boolean flag when expecting error without specifing the message it would be helpful. What do you think?